pyproject.toml

```diff
@@ -0,0 +1,106 @@
+[build-system]
+requires = ["hatchling>=1.25"]
+build-backend = "hatchling.build"
+
+[project]
+name = "model_signing"
+dynamic = ["version"]
+description = "A tool for signing and verifying ML models"
+readme = "README.model_signing.md"  # TODO: Merge READMEs and update
+license = { file = "LICENSE" }
+authors = [
+  { name = "Sigstore Authors", email = "sigstore-dev@googlegroups.com" }
+]
+classifiers = [
+  "License :: OSI Approved :: Apache Software License",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Development Status :: 3 - Alpha",
+  "Intended Audience :: Developers",
+  "Topic :: Security",
+  "Topic :: Security :: Cryptography",
+  "Topic :: Scientific/Engineering :: Artificial Intelligence",
+  "Typing :: Typed",
+]
+dependencies = [
+  "cryptography",  # TODO: make pki a feature
+  "in-toto-attestation",
+  "sigstore",
+  "typing_extensions",
+]
+requires-python = ">=3.11"
+keywords = [
+  "machine learning",
+  "artificial intelligence",
+  "model signing",
+  "AI security",
+  "ML security",
+  "AI supply chain security",
+  "ML supply chain security",
+]
+
+[project.urls]
+Homepage = "https://pypi.org/project/model-signing/"
+Changelog = "https://github.com/sigstore/model-transparency/blob/main/CHANGELOG.md"
+# TODO: https://github.com/sigstore/model-transparency/pull/279 - Documentation = "...."
+Source = "https://github.com/sigstore/model-transparency"
+Issues = "https://github.com/sigstore/model-transparency/issues"
+PyPI = "https://pypi.org/project/model-signing/"
+
+[tool.hatch.version]
+path = "src/model_signing/__init__.py"
+
+[tool.hatch.build]
+packages = ["src/model_signing"]
+
+[tool.hatch.envs.hatch-test]
+installer = "pip"
+parallel = true
+randomize = true
+
+[[tool.hatch.envs.hatch-test.matrix]]
+python = ["3.11", "3.12"]
+
+[tool.hatch.envs.type]
+description = """Custom environment for pytype.
+Use `hatch run type:check` to check types.
+"""
+extra-dependencies = [
+  "pytest",
+  "pytype",
+]
+installer = "pip"
+python = "3.11"
+
+[tool.hatch.envs.type.scripts]
+check = "pytype -k -j auto src tests"
+
+[tool.ruff]
+line-length = 80
+
+[tool.ruff.format]
+docstring-code-format = true
+line-ending = "lf"
+skip-magic-trailing-comma = true
+
+[tool.ruff.lint]
+select = ["B", "D", "E", "F", "I", "N", "PLC", "PLE", "PT", "SIM", "W"]
```

Comment:
should this be `extend-select`?
https://hatch.pypa.io/1.9/config/static-analysis/#extending-config

Or should we be explicit through `select`?

Reply:
We'll switch to `extend-select` once we minimize the rules we exclude, but right now we have a superset of what's by default, so `select` seems better.